### ISSUE TITLE
Scroll bug when presenting addCardViewController

### ISSUE THREAD

**Issue by Klingefjord** [stripe/stripe-ios#942]
## Summary

Scroll bug on addCardViewController. When textfield becomes first responder, the view scrolls autmoatically so that the card animation goes underneath the navigation bar

## Code to reproduce

This is what I use to present the addCardViewController:

```
    func initializeStripe(_ sender: UIViewController, key: String) {
        let addCardViewController = STPAddCardViewController(configuration: StripeUtil.getPaymentConfiguration(key),
                                                             theme: StripeUtil.getTheme())
        addCardViewController.delegate = self
        sender.present(UINavigationController(rootViewController: addCardViewController), animated: true)
        }
    }
```

## iOS version

11.3

## Installation method

CocoaPods

## SDK version

13.0.1

## Other information

![gid](https://user-images.githubusercontent.com/22666776/40348093-d58546ce-5da2-11e8-9010-ddc957a73f00.gif)

**Comment by joeydong-stripe** [comment]
Hi @Klingefjord, thanks for opening an issue and providing the gif.

This looks like the expected behavior as far as I know. The intended behavior is for the text field to be vertically center in the remaining space not taken up by the keyboard. As a result, the card can end up offscreen. What do you think?

Here is the code if you want to try experimenting with it: 

https://github.com/stripe/stripe-ios/blob/master/Stripe/UIViewController%2BStripe_KeyboardAvoiding.m#L50-L60

One option is to stop calling `stp_beginObservingKeyboardAndInsettingScrollView` but that could become an issue in this particular case if the screen is shorter and the text field can end up behind the keyboard frame.

**Comment by Klingefjord** [comment]
Personally I think the textfield shouldn't get centered as it looks especially weird during the card flip animation. Maybe some tweaking to the offset if the textfield is the first one underneath the card? I tried changing https://github.com/stripe/stripe-ios/blob/master/Stripe/UIViewController%2BStripe_KeyboardAvoiding.m#L111 to ```offset.y = ((topOfScreenOffset + topOfKeyboardOffset) / 2) - scrollView.contentInset.top - 70;```, It looks like the following on iPhone 5 and iPhone 7:

<img width="328" alt="skarmavbild 2018-05-23 kl 09 41 27" src="https://user-images.githubusercontent.com/22666776/40410901-30183ba6-5e6f-11e8-8fb4-c35c733b350a.png">

<img width="358" alt="skarmavbild 2018-05-22 kl 10 08 57" src="https://user-images.githubusercontent.com/22666776/40410950-5185d500-5e6f-11e8-8013-b8c49333b4be.png">

**Comment by joeydong-stripe** [comment]
Hi @Klingefjord 

Thanks for taking a look, you were on the right track. We had a team discussion and decided to change our opinion and try to have the scroll view behave as natively as possible.

I just opened this pull request https://github.com/stripe/stripe-ios/pull/943 and would love for you to try it out. As I mentioned in the pull request, we don't have too much context on how this class was implemented so we should invest extra time in testing the change.
